pkg/network/netlink/consumer.go

```diff
@@ -439,6 +444,10 @@ ReadLoop:
 			msgs = msgs[:len(msgs)-1]
 		}
 
+		if netns == 0 {
```

Comment:
If `netns != 0`, it won't be the inode number for the ns; I believe it is instead the ns id. In that case you will not use the `ns` passed in? 

I would just use the passed in `ns` always if `ns != 0`.

Reply:
We should probably be translating the ns id to the inode number then, otherwise we are mixing the two. Strangely, I have yet to get this feature of netlink to work for me. It never returns the ns values in the ancillary data.

Reply:
>  NETLINK_LISTEN_ALL_NSID (since Linux 4.2)
              When set, this socket will receive netlink notifications
              from all network namespaces that have an nsid assigned
              into the network namespace where the socket has been
              opened.  The nsid is sent to user space via an ancillary
              data.

This effectively means that we only listen to namespaces that are connected to the root with a veth, e.g. 

For dump, this flag does not come into play I believe, which is why you won't see any ns id's getting returned.

As for going from ns id to ino, that is not straightforward, i.e. there is no direct call to do that. We would have to maintain some sort of table with these mappings ourselves.

Reply:
Given all this, I'm going to ignore the ids coming from netlink until we have a solution to map between the two.